store/sqlstore/post_store.go

```diff
@@ -1735,6 +1735,39 @@ var specialSearchChar = []string{
 	":",
 }
 
+// GetRecentNthPostTime returns CreateAt time of most recent nth user-post
+func (s *SqlPostStore) GetRecentNthPostTime(n int64) (int64, error) {
+	if n <= 0 {
+		return 0, nil
+	}
+
+	builder := s.getQueryBuilder().
+		Select("CreateAt").
+		From("Posts p").
+		// Consider users posts only for cloud limit
+		Where(sq.And{
+			sq.Eq{"p.Type": ""},
+			sq.Expr("p.UserId NOT IN (SELECT UserId FROM Bots)"),
+		}).
+		OrderBy("p.CreateAt DESC").
+		Limit(1).
+		Offset(uint64(n - 1))
+
+	query, queryArgs, err := builder.ToSql()
+	if err != nil {
+		return 0, errors.Wrap(err, "GetRecentNthPostTime_tosql")
+	}
+
+	query = "SELECT COALESCE((" + query + "), 0)"
```

Comment:
I am not sure if this even works. Have you tested this locally? In any case, use COALESCE on the `createAt` column not on the whole query.

Reply:
Yes, the purpose of Coalesce here is to return 0 in case there is no row returned from the main query, which would happen when "offset" value is greater than the total no. of posts.
That would mean, all posts are still accessible/within the cloud limit.

Reply:
Can you post the final query that's being generated? Maybe I missed something.

Reply:
```sql
SELECT COALESCE((SELECT CreateAt FROM Posts p WHERE (p.Type = ? AND p.UserId NOT IN (SELECT UserId FROM Bots)) ORDER BY p.CreateAt DESC LIMIT 1 OFFSET 99), 0)
```